Core/src/org/sleuthkit/autopsy/communications/VisualizationPanel.java

```diff
@@ -142,12 +147,38 @@
     private final Map<NamedGraphLayout, JButton> layoutButtons = new HashMap<>();
     private NamedGraphLayout currentLayout;
 
+    @NbBundle.Messages("VisalizationPanel.paintingError=Problem painting visualization.")
     public VisualizationPanel() {
         initComponents();
+        //initialize invisible JFXPanel that is used to show JFXNotifications over this window.
+        notificationsJFXPanel.setScene(new Scene(new Pane()));
 
         graph = new CommunicationsGraph(pinnedAccountModel, lockedVertexModel);
 
-        graphComponent = new mxGraphComponent(graph);
+        /*
+         * custom implementation of mxGraphComponent that uses... a custom
+         * implementation of mxGraphControl ... that overrides paint so we can
+         * catch the NPEs we are getting and deal with them. For now that means
+         * just ignoring them.
+         */
+        graphComponent = new mxGraphComponent(graph) {
+            @Override
+            protected mxGraphComponent.mxGraphControl createGraphControl() {
+
+                return new mxGraphControl() {
+
+                    @Override
+                    public void paint(Graphics graphics) {
+                        try {
+                            super.paint(graphics);
+                        } catch (NullPointerException ex) { //NOPMD  We can't find the underlying cause in jgraphx
+                            logger.log(Level.WARNING, "There was a NPE while painting the VisualizationPanel", ex);
```

Comment:
What does NOPMD mean? Why are NPEs acceptable here? Please add some more explanation.

Reply:
NOPMD disables the warning that EeasyPMD would normaly show.  It is similar to the NO-NLS comment for internationalization.  I am not sure if codacy honors it.

I will improve the comment